k2/python/csrc/torch/fsa_algo.cu

```diff
@@ -661,6 +661,48 @@ static void PybindFixFinalLabels(py::module &m) {
      )");
 }
 
+static void PybindCtcGraph(py::module &m) {
+  m.def(
+      "ctc_graph",
+      [](const std::vector<std::vector<int32_t>> &symbols,
+         bool need_arc_map = true, int32_t gpu_id = -1)
+        -> std::pair<FsaVec, torch::optional<torch::Tensor>> {
+        ContextPtr context;
+        if (gpu_id < 0)
+          context = GetCpuContext();
+        else
+          context = GetCudaContext(gpu_id);
+
+        DeviceGuard guard(context);
+        Ragged<int32_t> ragged = CreateRagged2<int32_t>(symbols).To(context);
+        Array1<int32_t> arc_map;
+        FsaVec graph = CtcGraphs(ragged, need_arc_map ? &arc_map : nullptr);
+        torch::optional<torch::Tensor> tensor;
+        if (need_arc_map) tensor = ToTorch(arc_map);
+        return std::make_pair(graph, tensor);
+      },
+      py::arg("symbols"), py::arg("need_arc_map") = true,
+      py::arg("gpu_id") = -1,
+      R"(
+  If gpu_id is -1, the returned FsaVec is on CPU.
+  If gpu_id >= 0, the returned FsaVec is on the specified GPU.
+      )");
+
+  m.def(
+      "ctc_graph",
+      [](const Ragged<int32_t> &symbols, bool need_arc_map = true,
+         int32_t /*unused_gpu_id*/)
```

Comment:
The last argument has no default value but it is after an argument with default value.
It's not valid C++, I think.

Reply:
I saw the error in actions, it could compile successfully on linux, will fix it.